pkg/generate/app/pipeline.go

```diff
@@ -456,6 +468,97 @@ func NewAcceptUnique(typer runtime.ObjectTyper) Acceptor {
 	}
 }
 
+type acceptNonExistentImageStream struct {
+	typer     runtime.ObjectTyper
+	getter    imageclient.ImageInterface
+	namespace string
+}
+
+// Accept accepts any non-ImageStream object or an ImageStream that does
+// not exist in the api server
+func (a *acceptNonExistentImageStream) Accept(from interface{}) bool {
+	obj, _, err := objectMetaData(from)
+	if err != nil {
+		return false
+	}
+	gvk, _, err := a.typer.ObjectKinds(obj)
+	if err != nil {
+		return false
+	}
+	gk := gvk[0].GroupKind()
+	if !image.IsKindOrLegacy("ImageStream", gk) {
+		return true
+	}
+	is, ok := from.(*image.ImageStream)
+	if !ok {
+		glog.V(4).Infof("type cast to image stream %#v not right for an unaticipated reason", from)
+	}
+	imgstrm, err := a.getter.ImageStreams(a.namespace).Get(is.Name, metav1.GetOptions{})
+	if err == nil && imgstrm != nil && imgstrm.Name == is.Name {
```

Comment:
imgstrm.Name==is.Name check seems redundant, did you find it necessary?

Reply:
This is the piece regarding the odd behavior around getting image streams in the integration tests I referred to in my recent comment ... yes, this should not be necessary, but instead of nil I was getting this empty imagestream back ... I plan on asking the appropriate folks about this and will report back